app/scripts/lib/transaction/swaps.ts

```diff
@@ -0,0 +1,122 @@
+import { pickBy, merge } from 'lodash';
+import {
+  TransactionStatus,
+  TransactionType,
+  TransactionMeta,
+} from '../../../../shared/constants/transaction';
+import MetamaskController from '../../metamask-controller';
+
+type SwapOptions = {
+  // Whether this transaction required an approval transaction
+  hasApproveTx: boolean;
+  // Additional metadata to store for the transaction
+  meta: any;
+};
+
+type SwapApprovalTransaction = {
+  sourceTokenSymbol: string;
+  destinationTokenSymbol: string;
+  type: TransactionType;
+  destinationTokenDecimals: string;
+  destinationTokenAddress: string;
+  swapMetaData: string;
+  swapTokenValue: string;
+  estimatedBaseFee: string;
+  approvalTxId: string;
+};
+
+/**
+ * Creates a new transaction with swap-specific metadata
+ *
+ * @param this - MetamaskController
+ * @param swapOptions - Swap-specific options
+ * @param transactionType - The type of transaction
+ * @param transactionMeta - The transaction metadata
+ * @returns The updated transaction metadata
+ */
+export async function createSwapsTransaction(
+  this: MetamaskController,
+  swapOptions: SwapOptions,
+  transactionType: TransactionType,
+  transactionMeta: TransactionMeta,
+) {
+  if (
```

Comment:
For clarity sake, can we include the original large comment here to explain why this logic is needed?

Reply:
Whole function with comments rolled back in https://github.com/MetaMask/metamask-extension/pull/21456/commits/86dfac2fc9acf571f3240e2ea59948f8ed6b4a1f